tests/baselines/reference/controlFlowGenericTypes.types

```diff
@@ -160,8 +160,8 @@ function g1<T extends Box<T> | undefined>(x: T) {
 >         : ^^^^^^^
 >isBox : (x: any) => x is Box<unknown>
 >      : ^ ^^   ^^^^^                 
->x : Box<T> | undefined
->  : ^^^^^^^^^^^^^^^^^^
+>x : T

```

Comment:
all the baseline changes like this are caused by the fact that now `hasContextualTypeWithNoGenericTypes` returns `false` when the contextual type is any, and thus `substituteConstraints` gets computed as `false` in `getNarrowableTypeForReference`